sapphire/analysis/reconstructions.py

```diff
@@ -447,3 +441,17 @@ def _store_reconstruction(self, coincidence, core_x, core_y, theta, phi,
             row['s%d' % number] = True
 
         row.append()
+
+    def _get_active_stations(self):
+        """Return station numbers with non-empty event table in datafile"""
+        def stations_with_events():
```

Comment:
I'll be damned:
```python
>>> def f(N):
    l = []
    for i in xrange(N):
        l.append(i)
    return l
...
>>> def g(N):
    def h(N):
        for i in xrange(N):
            yield i
    return list(h(N))
...
>>> f(100000) == g(100000)
True
>>> %timeit f(100000)
10 loops, best of 3: 15.4 ms per loop
>>> %timeit g(100000)
100 loops, best of 3: 10.1 ms per loop
```
I'll admit I don't understand your outside-of-the-VM argument.